airflow/models/trigger.py

```diff
@@ -241,8 +241,9 @@ def assign_unassigned(cls, triggerer_id, capacity, heartrate, session: Session =
     def get_sorted_triggers(cls, capacity, alive_triggerer_ids, session):
         return with_row_locks(
             session.query(cls.id)
+            .join(TaskInstance, cls.id == TaskInstance.trigger_id, isouter=True)
             .filter(or_(cls.triggerer_id.is_(None), cls.triggerer_id.notin_(alive_triggerer_ids)))
-            .order_by(cls.created_date)
+            .order_by(-TaskInstance.priority_weight, cls.created_date)
```

Comment:
```suggestion
            .order_by(TaskInstance.priority_weight.desc(), cls.created_date)
```

Preferred style

Reply:
For NULL, I can use `TaskInstance.priority_weight.desc().nulls_last()`, but it seems like the lock (with_for_update) doesn't work with the outer join, I will try to find a solution for that.

Reply:
Can we use `CASE` instead? We don’t really need to put them last, just probably not at the very beginning.